src/edg.rs

```diff
@@ -350,11 +453,26 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
 
     // Mark `dependency` as a prerequisite for finding the final assignment of `vertex`
     fn add_depend(&mut self, vertex: &V, dependency: Edges<V>) {
+        // Update the distance
+        let default = 0;
+        let tdist = self.distances.get(vertex).unwrap_or(&default).clone();
+        match dependency.clone() {
+            Edges::NEGATION(edge) => {
+                let sdist = self.distances.get(&edge.source).unwrap_or(&default).clone() + 1;
+                self.distances.insert(vertex.clone(), max(sdist, tdist));
+            }
+            Edges::HYPER(edge) => {
+                let sdist = self.distances.get(&edge.source).unwrap_or(&default).clone();
+                self.distances.insert(vertex.clone(), max(sdist, tdist));
+            }
+        }
+
+        // Mark `dependency` as a prerequisite for finding the final assignment of `vertex`
         if let Some(dependencies) = self.depends.get_mut(vertex) {
-            dependencies.insert(dependency);
+            dependencies.insert(dependency.clone());
```

Comment:
Fixes `clippy` warning
```suggestion
            dependencies.insert(dependency);
```